### ISSUE TITLE
I am very sorry that I found the dimension of probability matrix wrong.

### ISSUE THREAD

**Issue by qinwenchuang** [pgmpy/pgmpy#1532]
### Subject of the issue
Describe your issue here.
I am very sorry that I found the dimension of probability matrix wrong when I was training BN model.
For example, a node with five states has two node inputs, each of which has five states.
Theoretically, the probability matrix dimension of this node is 5*10, but the result is 4*10, but the sum of the columns of the matrix is 1.0.
So, I guess get_cpds() is ignored if a row is all 0.0?
In order to avoid dimension verification and extra calculation in C++ environment, can all the results be output as expected, and all the rows in time

### Your environment
* pgmpy version:0.1.18
* Python version:3.8
* Operating System:ubuntu20.04

### Steps to reproduce
Tell us how to reproduce this issue. Please provide a **minimal reproducible code** of
the issue you are facing if possible.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

**Comment by qinwenchuang** [comment]
Theoretically, the probability matrix dimension of this node is 5 * 10, but the result is 4 * 10, but the sum of the columns of the matrix is 1.0.

**Comment by ankurankan** [comment]
@qinwenchuang If I am understanding correctly, you have some states which are not observed in your dataset. If you want to force pgmpy to create CPDs including states which are not present in the data, you can specify a `state_names` argument in the `fit` method with all the possible states for the variables. Here's an example:
```python
# Create a random dataset
> from pgmpy.utils import get_example_model
> model =  get_example_model('asia')
> data = model.simulate(int(1e4))

# All the variables in this dataset have two states yes and no. 
# Let's assume that the `smoke` variable should have a third state called `missing` which is not observed in the dataset.
> from pgmpy.models import BayesianNetwork
> new_model = BayesianNetwork(model.edges())
> new_model.fit(data, state_names={'smoke': ['no', 'yes', 'missing']})
> print(new_model.get_cpds('smoke'))
> print(new_model.get_cpds('lung'))

Generating for node: dysp: 100%|███████████████████████████████████████████| 8/8 [00:00<00:00, 88.04it/s]
+----------------+--------+
| smoke(no)      | 0.5039 |
+----------------+--------+
| smoke(yes)     | 0.4961 |
+----------------+--------+
| smoke(missing) | 0      |
+----------------+--------+

+-----------+--------------------+---------------------+----------------+
| smoke     | smoke(no)          | smoke(yes)          | smoke(missing) |
+-----------+--------------------+---------------------+----------------+
| lung(no)  | 0.9921765295887663 | 0.8993020937188435  | 0.5            |
+-----------+--------------------+---------------------+----------------+
| lung(yes) | 0.0078234704112337 | 0.10069790628115653 | 0.5            |
+-----------+--------------------+---------------------+----------------+

```

**Comment by qinwenchuang** [comment]
It works. From network building to fitting, the number of node states is not set. If the data is insufficient, the loss of some states of the nodes is expected.
Thanks for the guidance.
